kore/src/Kore/Internal/Substitution.hs

```diff
@@ -3,7 +3,7 @@ Copyright   : (c) Runtime Verification, 2018-2021
 License     : BSD-3-Clause
 -}
 module Kore.Internal.Substitution (
-    Substitution,
+    Substitution (..),
```

Comment:
Please don't export the constructors, they're not exported on purpose.

I'd suggest implementing the logic you need in a separate function in this module, and export that to be used in the desired place.